src/core/XHPUnsafeAttributeValue.php

```diff
@@ -0,0 +1,24 @@
+<?hh // strict
+/*
+ *  Copyright (c) 2004-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the MIT license found in the
+ *  LICENSE file in the root directory of this source tree.
+ *
+ */
+
+/**
+ * INCREDIBLY DANGEROUS: Marks an object as being able to provide an HTML
+ * string.
+ *
+ * This is useful for attribute values that contain html entities which should
+ * not be escaped, such as an onload attribute which can contain a snippet of JS.
```

Comment:
in general, this should still be escaped as appropriate for *attribute values* (e.g. '"' should be escaped) - are we just doing an XSS-safe-but-wrong escaping here?

For your case, this might be another bug, not actually a need for unescaped values.

Reply:
Coming back to this. Are you saying that perhaps `renderBaseAttrs()` should just do this:

```
$buf .= ' '.
            htmlspecialchars($key).
            '="'.
            \HH\Lib\Str\replace($value, '"', '&quot;').
            '"';
        }
```

That does seem like it'd solve the problem without creating the possibility of breaking the HTML for strings with double quotes in them.

Reply:
I see that HSL isn't currently vendored in this repo so perhaps that'd just be `str_replace` for now but same idea.

Reply:
Looking at the spec, what we're already doing is correct:

XHP's job can be described as turning `T` into `AttributeEscaped<T>`
In this case, we can imagine there's additional types:
- `JSString`
- `AttributeEscaped<JSString>`

... and for onload, XHP should turn a `JSString` into an `AttributeEscaped<JSString>`, and it is doing that correctly. However, for your migration it sounds like you already have `AttributeEscaped<JSString>` - so XHP will produce `AttributeEscaped<AttributeEscaped<JSString>>`.

Providing `XHPUnsafeAttributeValue` seems the correct fix here, with no other changes; what I'd like to see is changes to the comment and test so that they don't imply that they're needed/appropriate for a `JSString` - only if you already have an AttributeEscapedString

Reply:
Ok, this makes sense to me and I've updated the PR. Thanks for helping me through this!